src/phoenix/experimental/evals/models/openai.py

```diff
@@ -229,6 +271,38 @@ def _generate(self, prompt: str, **kwargs: Any) -> str:
             return str(function_call.get("arguments") or "")
         return str(message["content"])
 
+    async def _async_generate_with_retry(self, **kwargs: Any) -> Any:
+        """Use tenacity to retry the completion call."""
+        openai_retry_errors = [
+            self._openai.APITimeoutError,
+            self._openai.APIError,
+            self._openai.APIConnectionError,
+            self._openai.RateLimitError,
+            self._openai.InternalServerError,
+        ]
+
+        @self.retry(
+            error_types=openai_retry_errors,
+            min_seconds=self.retry_min_seconds,
+            max_seconds=self.retry_max_seconds,
+            max_retries=self.max_retries,
+        )
+        async def _completion_with_retry(**kwargs: Any) -> Any:
+            if self._model_uses_legacy_completion_api:
+                if "prompt" not in kwargs:
+                    kwargs["prompt"] = "\n\n".join(
+                        (message.get("content") or "")
+                        for message in (kwargs.pop("messages", None) or ())
+                    )
+                # OpenAI 1.0.0 API responses are pydantic objects, not dicts
+                # We must dump the model to get the dict
+                res = await self._async_client.completions.create(**kwargs)
+            else:
+                res = await self._async_client.chat.completions.create(**kwargs)
+            return res.model_dump()
+
+        return await _completion_with_retry(**kwargs)
```

Comment:
Is the plan to re-queue failed calls on the queue as a follow-up?

Reply:
Nothing here gets requeued, this is our existing retry mechanism and runs inside the executor